GoogleSignIn/Sources/GIDGoogleUser.m

```diff
@@ -116,6 +116,11 @@ - (void)encodeWithCoder:(NSCoder *)encoder {
   [encoder encodeObject:_hostedDomain forKey:kHostedDomainKey];
 }
 
+#pragma mark - private method
+- (void)setScopes:(NSArray<NSString *> *)scopes {
```

Comment:
Note that the OIDAuthState and GIDProfileData objects that back the GIDGoogleUser instance are changing in the the addScopes scenario, so you'll want to update more than just the scopes property here.

Reply:
Update the implementation according to Peter's suggestion.